core/autocrypt/crypto.py

```diff
@@ -0,0 +1,574 @@
+                compression=[CompressionAlgorithm.ZLIB,
+                             CompressionAlgorithm.BZ2,
+                             CompressionAlgorithm.ZIP,
+                             CompressionAlgorithm.Uncompressed])
+        logger.debug('uid added')
+        if add_subkey is True:
+            subkey = PGPKey.new(alg_subkey, size)
+            logger.debug('new subkey')
+            skey.add_subkey(subkey,
+                            usage={KeyFlags.EncryptCommunications,
+                                   KeyFlags.EncryptStorage})
+            logger.debug('subkey added')
+        if protected is True:
+            passphrase = getpass.getpass()
+            skey.protect(
+                             passphrase,
+                             SymmetricKeyAlgorithm.AES256,
+                             HashAlgorithm.SHA256)
+            logger.debug('Key protected')
+        # NOTE: this is not needed, as it it signed by default
+        # self.sign_own_key()
+
+        # put the key as exported ASCII-armored in ring
+        self.export_key(skey)
+        logger.debug('exported key')
+        self.publicpgpykeys.append(skey.pubkey)
+        self.secretpgpykeys.append(skey)
+        logger.debug('updated loaded keys')
+        logger.debug('self.publicpgpykeys %s', self.publicpgpykeys)
+        logger.debug('self.secretpgpykeys %s', self.secretpgpykeys)
+        # self.load_pgpykr()
+        # self.add_key(skey)
+        # self.load_keys_from_pgpykr()
+        if _own is True:
+            self.own_pgpykey = skey
+            self.own_keyhandle = skey.fingerprint.keyid
+            logger.debug('longid %s', self.own_keyhandle)
+        return skey.fingerprint.keyid
+
+    def supports_eddsa(self):
+        # NOTE: PGPy does not currently support it
+        return False
+
+    def export_skey(self, pgpykey=None):
+        if pgpykey is None:
+            pgpykey = self.own_pgpykey
+        assert type(pgpykey) == PGPKey
+        assert not pgpykey.is_public
+        secretkeydata = self.get_secret_keydata(armor=True,
+                                                pgpykey=pgpykey)
+        skpath = os.path.join(self.pgpydir, pgpykey.fingerprint.keyid
+                              + '.key')
+        with open(skpath, 'w') as fd:
+            fd.write(secretkeydata)
+            logger.debug('written %s', skpath)
+
+    def export_pkey(self, pgpykey=None):
+        if pgpykey is None:
+            pgpykey = self.own_pgpykey.pubkey
+        assert type(pgpykey) == PGPKey
+        if not pgpykey.is_public:
+            pgpykey = pgpykey.pubkey
+        publickeydata = self.get_public_keydata(armor=True,
+                                                pgpykey=pgpykey)
+        pkpath = os.path.join(self.pgpydir, pgpykey.fingerprint.keyid
+                              + '.asc')
+        with open(pkpath, 'w') as fd:
+            fd.write(publickeydata)
+            logger.debug('written %s', pkpath)
```

Comment:
why does exporting a public key perform a write operation on the file system here? 
Also "export_pkey" is ambigious as it could mean "export_private_key" and only is clearer when one also knows about "export_skey" and figures out the "s" means secret.